### DISCUSSION TITLE
Where is state stored?

### DISCUSSION THREAD

**Discussion by nephix** [preactjs/signals#505] [General]
Hey guys 👋 

where is the actual state of a signal stored, does it only exist in memory?

Background to my question is that we have multiple web components that access signal state, but we noticed it's shared across the components. 

Can 3rd party scripts (eg embedded on the site) access it?

**Comment by rschristian** [comment]
> where is the actual state of a signal stored, does it only exist in memory?

Of course. You can set up an effect to save it to local storage (or an alternative) if you wish.

> Background to my question is that we have multiple web components that access signal state, but we noticed it's shared across the components.

Unless you localize the signal to the component instance, this will of course be the case. It's no different from any other value in that regard. If it's outside a component, and multiple components pull from it, they're going to be sharing state.

> Can 3rd party scripts (eg embedded on the site) access it?

If you attach it to the window/make it global, sure. Otherwise, no.
